mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+# -*- encoding: utf-8 -*-
+'''Display functions'''
+
+import numpy as np
+from scipy.signal import spectrogram
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle, Patch
+from matplotlib.ticker import FuncFormatter, MultipleLocator
+from matplotlib.colors import LinearSegmentedColormap, LogNorm
+
+from .melody import freq_to_voicing
+from .util import midi_to_hz, hz_to_midi
+
+
+def __expand_limits(ax, limits, which='x'):
+    '''Helper function to expand axis limits'''
+
+    if which == 'x':
+        getter, setter = ax.get_xlim, ax.set_xlim
+    elif which == 'y':
+        getter, setter = ax.get_ylim, ax.set_ylim
+    else:
+        raise ValueError('invalid axis: {}'.format(which))
+
+    old_lims = getter()
+    new_lims = list(limits)
+
+    if np.isfinite(old_lims[0]):
```

Comment:
This should only happen for freshly constructed axes with no data attached.  I'll add a comment.